### PR TITLE
WIP: updates for sbpy.data

### PR DESCRIPTION
**Pull request by mommermi**
* enable `start`-`number`-`stop` epochs for `ephem.from_horizons`
* added `EarthLocation` support for `ephem.from_horizons`
* added functionality to query multiple targets with `ephem.from_mpc`
* added `Ephem.from_miriade`
* all query functions now require `Time` objects as epochs
* time scale information in `Time` objects is now fully accounted for
* implemented `orbit.from_mpc`
* implemented `obs.from_mpc` and `obs.supplement`

### CODE REVIEW

File: sbpy/data/ephem.py [hunk 1]
```diff
@@ -143,11 +178,17 @@ def from_horizons(cls, targetids, id_type='smallbody',
...
+                                scale='utc')
+        if any(all_eph['datetime_jd'] < 2437665.5):
+            print('pimmel')
+            all_eph['epoch'][all_eph['datetime_jd'] < 2437665.5] = (
+                Time(all_eph['datetime_jd'][all_eph['datetime_jd'] <
+                                            2437665.5],
+                     format='jd', scale='ut1'))
+        all_eph.remove_column('datetime_jd')
 
         if bib.status() is None or bib.status():
```
Comment (mkelley) [inline sbpy/data/ephem.py]: This test is already addressed in `bib.register`, so no need to have it here.  It occurs 6 times in `sbpy.data`.

File: sbpy/data/core.py [hunk 2]
```diff
@@ -135,15 +149,15 @@ def from_dict(cls, data, meta={}, **kwargs):
         Examples
         --------
         The following example creates a single-row `~sbpy.data.Orbit`
-        object (the other `~DataClass` objects work the exact same way).
+        object(the other `~DataClass` objects work the exact same way).
 
-        >>> import astropy.units as u
-        >>> from sbpy.data import Orbit
-        >>> orb = Orbit.from_dict({'a': 2.7674*u.au,
+        >> > import astropy.units as u
```
[the remark this answers is not in the document]
Comment (mommermi) [reply sbpy/data/core.py]: Emacs does this every once in a while... fixed it, thanks!

File: sbpy/data/core.py [hunk 3]
```diff
@@ -176,22 +190,22 @@ def from_dict(cls, data, meta={}, **kwargs):
         requires a specific order, use ``OrderedDict``. This example
         also shows the use of meta data.
 
-        >>> from collections import OrderedDict
-        >>> orb = Orbit.from_dict(OrderedDict([('a', [2.7674, 3.123]*u.au),
+        >> > from collections import OrderedDict
+        >> > orb = Orbit.from_dict(OrderedDict([('a', [2.7674, 3.123]*u.au),
         ...                                    ('e', [0.0756, 0.021]),
         ...                                    ('i', [10.59, 3.21]*u.deg)]),
         ...                                   meta={'targetname': 'asteroid'})
```
[the remark this answers is not in the document]
Comment (mommermi) [reply sbpy/data/core.py]: fixed

File: sbpy/data/core.py [hunk 4]
```diff
@@ -135,15 +149,15 @@ def from_dict(cls, data, meta={}, **kwargs):
         Examples
         --------
         The following example creates a single-row `~sbpy.data.Orbit`
-        object (the other `~DataClass` objects work the exact same way).
+        object(the other `~DataClass` objects work the exact same way).
```
[the remark this answers is not in the document]
Comment (mommermi) [reply sbpy/data/core.py]: good catch!

File: sbpy/data/core.py [hunk 5]
```diff
@@ -220,25 +239,25 @@ def from_columns(cls, columns, names, units=None, meta={}, **kwargs):
...
+        names: str or list-like
             Field names, must have the same number of names as data columns.
-        units : str or list-like, optional
-            Unit labels (as provided by `~astropy.units.Unit`) in which
+        units: str or list-like, optional
+            Unit labels(as provided by `~astropy.units.Unit`) in which
             the data provided in ``columns`` will be stored in the underlying
             table. If None, the units as provided by ``columns``
             are used. If the units provided in ``units`` differ from those
             used in ``colums``, ``columns`` will be transformed to the units
```
[the remark this answers is not in the document]
Comment (mommermi) [reply sbpy/data/core.py]: another good one, thanks!

File: sbpy/data/orbit.py [hunk 6]
```diff
@@ -14,13 +14,19 @@
 from astropy.table import vstack, Column
 from astroquery.jplhorizons import Horizons
 import astropy.units as u
+from warnings import warn
 
 from .. import bib
+from ..exceptions import SbpyException, SbpyWarning
```
[the remark this answers is not in the document]
Comment (mommermi) [reply sbpy/data/orbit.py]: fixed this, thanks!

**Pull request merged**